src/App.tsx

```diff
@@ -1,24 +1,161 @@
+import { TodoNotification }
+  from './components/TodoNotification/TodoNotification';
+import { createTodo, getTodos, removeTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterOptions } from './types/FilterOptions';
+import { filterTodos } from './Helpers';
+import { ErrorMessages } from './types/ErrorMessages';
 
-const USER_ID = 0;
+const USER_ID = 6795;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterOption, setFilterOption] = useState(FilterOptions.ALL);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState<ErrorMessages | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodosId, setDeletingTodosId] = useState<number[]>([]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessages.LoadError);
+      });
+  }, []);
+
+  const visibleTodos = useMemo(() => filterTodos(todos, filterOption),
+    [filterOption, todos]);
+
+  const activeVisibleTodos = useMemo(
+    () => filterTodos(todos, FilterOptions.ACTIVE),
+    [todos],
+  );
+  const completedVisibleTodos = useMemo(
+    () => filterTodos(todos, FilterOptions.COMPLETED),
+    [todos],
+  );
+  const isTodosPresent = todos.length > 0;
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const handleNewTodoTitleChange
+  = (event: React.ChangeEvent<HTMLInputElement>) => (
+    setNewTodoTitle(event.target.value)
+  );
+
+  const clearNewTodoTitle = () => (
+    setNewTodoTitle('')
+  );
+
+  const addNewTodo = async (title: string) => {
+    if (!title) {
+      setErrorMessage(ErrorMessages.TitleError);
+
+      return null;
+    }
+
+    try {
+      const newTodoPayload = {
+        completed: false,
+        title,
+        userId: USER_ID,
+      };
+
+      setTempTodo({
+        id: 0,
+        ...newTodoPayload,
+      });
+
+      const newTodo = await createTodo(newTodoPayload);
+
+      clearNewTodoTitle();
+      setTodos((prevState) => [...prevState, newTodo]);
+
+      return newTodo;
+    } catch (error) {
+      setErrorMessage(ErrorMessages.TitleError);
+
+      return null;
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const deleteTodo = async (todoId: number) => {
+    try {
+      setDeletingTodosId((prevState) => [...prevState, todoId]);
+
+      const removeResult = await removeTodo(todoId);
+
+      if (!removeResult) {
+        setErrorMessage(ErrorMessages.DeleteError);
+
+        return;
+      }
+
+      setTodos((prevState) => {
+        const resultState = prevState.filter(todo => todo.id !== todoId);
+
+        return [...resultState];
+      });
+    } catch (error) {
+      setErrorMessage(ErrorMessages.DeleteError);
+    } finally {
+      setDeletingTodosId([]);
+    }
+  };
+
+  const deleteTodos = (todosToDelete: Todo[]) => {
```

Comment:
it will be better to rename it to `clearCompletedTodos` of smth like this, because u've already have `deleteTodo` and this helps understand meaning of this function